src/fetchers/wakatime-fetcher.js

```diff
@@ -4,29 +4,30 @@ import { MissingParamError } from "../common/utils.js";
 /**
  * WakaTime data fetcher.
  *
- * @param {{username: string, api_domain: string, range: string}} props Fetcher props.
+ * @param {{username: string, api_domain: string}} props Fetcher props.
  * @returns {Promise<WakaTimeData>} WakaTime data response.
  */
-const fetchWakatimeStats = async ({ username, api_domain, range }) => {
+const fetchWakatimeStats = async ({ username, api_domain }) => {
   if (!username) throw new MissingParamError(["username"]);
 
-  try {
-    const { data } = await axios.get(
-      `https://${
-        api_domain ? api_domain.replace(/\/$/gi, "") : "wakatime.com"
-      }/api/v1/users/${username}/stats/${
-        range || "all_time"
-      }?is_including_today=true`,
-    );
-
-    return data.data;
-  } catch (err) {
-    if (err.response.status < 200 || err.response.status > 299) {
-      throw new Error(
-        "Wakatime user not found, make sure you have a wakatime profile",
+  // Loop through available ranges to get user data.
+  for (const range of ["all_time", "last_year", "last_7_days"]) {
```

Comment:
Instead of a loop you can also fetch https://wakatime.com/api/v1/users/current and use `public_profile_time_range` to find the user's public profile range. It might end up being the same performance, sometimes taking one request more and sometimes saving one request depending on how many tries before this loop finds the right range.

Reply:
@alanhamlett I agree using that endpoint is better. Unfortunately, last time I checked, Wakatime decided to put this endpoint behind an auth wall 😓. When I try to fetch `https://wakatime.com/api/v1/users/rickstaa` from within GRS for example, I get the following error:

```json
{"error": "Unauthorized"}
```

I think the only way to use this API is to add an authentication token.

Reply:
Oh yea, then your loop solution is the best way.

Reply:
Because of your comment, I looked into the API a bit more and found out that we can get the range from the https://wakatime.com/api/v1/users/rickstaa/stats. I created 6b8e06b2e40d74be58003cbb6897729a7fa1010d to apply your suggestions.